### DISCUSSION TITLE
Dependent Selection Code Not Working in Production Environment (Pimcore v10.5.21)

### DISCUSSION THREAD

**Discussion by KasthuriAnandhan** [pimcore/pimcore#17553] [Q&A / Support] [answered]
Hi @kingjia90 

The provided code for dependent selection based on the Brand attribute works in my local Pimcore environment (v11.3.1.1). However, when I tried the same code in the production Pimcore environment (v10.5.21), it didn’t work as expected. This might be due to differences between the two Pimcore versions.

Reference:
[OptionProvider.php.txt](https://github.com/user-attachments/files/16839181/OptionProvider.php.txt)
Classes:
Products: has Brand (Input) & Brands (One-to-One relation)
Brands: has BrandName (Input)
Category: has Category (Input) & Brand (One-to-Many relation)
Please advise on potential version-related issues or other factors that could cause this discrepancy.

Thanks

**Comment by kingjia90** [comment, marked as answer]
In Pimcore 11, the parameter and return types are more strict, maybe removing them would work
You can take the one from the old documentation as blueprint
https://pimcore.com/docs/pimcore/10.6/Development_Documentation/Objects/Object_Classes/Data_Types/Dynamic_Select_Types.html
![image](https://github.com/user-attachments/assets/562a753c-5931-4bce-b09c-68ec95c80e8b)

**Comment by KasthuriAnandhan** [comment]
Hi @kingjia90,

Following the old documentation did not resolve the issue. Is achieving this functionality also possible in Pimcore version 10.6 or above?

**Reply by kingjia90** [reply]
It may be worth to try it, other than that i don't know any other way to make it work.
Could you also share what shows as an error?

**Comment by KasthuriAnandhan** [comment]
Thank You @kingjia90.
